File: src/main/java/net/casualuhc/uhcmod/command/PosCommand.java
package net.casualuhc.uhcmod.command;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.context.CommandContext;
import com.mojang.brigadier.exceptions.CommandSyntaxException;
import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
import net.minecraft.entity.Entity;
import net.minecraft.scoreboard.AbstractTeam;
import net.minecraft.server.command.CommandManager;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.text.Text;
import net.minecraft.util.math.Vec3d;

import static net.minecraft.server.command.CommandManager.literal;

public class PosCommand {
    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(literal("pos").requires(source -> source.getEntity() != null).executes(PosCommand::sendPosition));
    }

    public static int sendPosition(CommandContext<ServerCommandSource> context) throws CommandSyntaxException {
        ServerCommandSource source = context.getSource();
        CommandManager manager = source.getServer().getCommandManager();

        Entity entity = source.getEntityOrThrow();
        Vec3d pos = entity.getPos();

        AbstractTeam abstractTeam = entity.getScoreboardTeam();

        if (abstractTeam != null) {
            manager.execute(source, String.format("tm I'm at %.0f, %.0f, %.0f", pos.x, pos.y, pos.z));
        }
        else {
            throw new SimpleCommandExceptionType(Text.literal("§c[ERROR] You can only run this command if you are alive and in a team!")).create();
        }
        return 0;
    }
}

File: src/main/kotlin/net/casual/commands/PosCommand.kt
package net.casual.commands

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import com.mojang.brigadier.exceptions.SimpleCommandExceptionType
import net.casual.arcade.utils.ComponentUtils.command
import net.casual.arcade.utils.ComponentUtils.lime
import net.casual.arcade.utils.PlayerUtils.teamMessage
import net.casual.util.Texts
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.minecraft.network.chat.Component

object PosCommand: Command {
    private val NO_TEAM = SimpleCommandExceptionType(Texts.COMMAND_NO_TEAM)

    override fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
            Commands.literal("pos").executes(this::execute)
        )
    }

    private fun execute(context: CommandContext<CommandSourceStack>): Int {
        val player = context.source.playerOrException
        val team = player.team

        if (team === null) {
            throw NO_TEAM.create()
        }

        val pos = player.position()
        val position = Component.literal("[%.0f, %.0f, %.0f]".format(pos.x, pos.y, pos.z))
            .lime()
            .command("/look towards ${pos.x} ${pos.y} ${pos.z}")

        player.teamMessage(Texts.COMMAND_POS.generate(position))
        return 1
    }
}